linalg/hypre.cpp

```diff
@@ -1856,6 +1857,11 @@ HypreParMatrix * HypreParMatrixFromBlocks(Array2D<HypreParMatrix*> &blocks,
                                                (*(blocks(i, j)));
                MFEM_ASSERT(parcsr_op != NULL, "const_cast failed");
                csr_blocks(i, j) = hypre_MergeDiagAndOffd(parcsr_op);
+#if MFEM_HYPRE_VERSION >= 21600
+               MFEM_VERIFY(csr_blocks(i, j)->num_rows < INT_MAX,"Number of "
+                           "local rows is too large to store as an integer.");
```

Comment:
1. Why not use `<=` here?
2. The `J` array stores column indices, shouldn't the check be for `num_cols`?

Reply:
Good points. I copied this from `superlu.cpp` without thinking about it. Should I make the same changes to that file?

Reply:
I think so. @mlstowell, what do you think?

Reply:
Another thing: when `HYPRE_Int` is 64bit, the check with `INT_MAX` is wrong -- there is no need to check anything.

Reply:
When @jandrej and I discussed this in his SuperLU PR #988 we were talking about checking `num_cols`.  I'm not sure why the test uses `num_rows`.  I must have written the wrong thing.

Reply:
In the SuperLU wrapper we needed an `int*` point to `j` to pass into a SuperLU matrix data structure so we had to first copy the 64bit `HYPRE_Ints` to `ints`.  In this case I don't think the copy is necessary.  You need to check whether HYPRE is storing the column indices in `hypre_CSRMatrix::j` or `hypre_CSRMatrix::BigJ` and read them from the correct array.

Reply:
@mlstowell In seq_mv.h it looks like there are arrays `j` and `big_j` in `hypre_CSRMatrix`. Does that mean one is `NULL` and the other is not, in every case? If so, I can just check for which one is not `NULL`.